lib/secure-hash.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+module.exports = function (base64, crypto, promise) {
+    crypto.pbkdf2Async = promise.promisify(crypto.pbkdf2);
+
+    /**
+     * Hashes the content passed using defaults or
+     * configuration options if passed in.
+     *
+     * @param {(Buffer|string)} data
+     * @param {Object} config
+     * @return {Promise.<Buffer>}
+     */
+    function secureHashAsync(data, config) {
+        var algorithm, hashLength, iterations, salt;
+
+        if (! data) {
+            throw new Error("Nothing to hash");
+        }
+
+        if (typeof data === "string") {
+            data = new Buffer(data, "binary");
+        }
+
+        config = config ? config : {};
```

Comment:
Ah yes, I meant to change this to the way above.  Missed that before I sent the pull request.